conformance/tests/mesh/httproute-rewrite-path.go

```diff
@@ -0,0 +1,155 @@
+/*
+Copyright 2025 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package meshtests
+
+import (
+	"testing"
+
+	"sigs.k8s.io/gateway-api/conformance/utils/echo"
+	"sigs.k8s.io/gateway-api/conformance/utils/http"
+	"sigs.k8s.io/gateway-api/conformance/utils/suite"
+	"sigs.k8s.io/gateway-api/pkg/features"
+)
+
+func init() {
+	MeshConformanceTests = append(MeshConformanceTests, MeshHTTPRouteRewritePath)
+}
+
+var MeshHTTPRouteRewritePath = suite.ConformanceTest{
```

Comment:
Not blocking just to think about. Over time, we will have more and more features in gateway. 99% of them are not mesh or gateway specific, yet here we are building a pattern of completely distinct tests for the two.

This test could be generic to mesh/gateway and share almost all the same code. There are only two differents needed:
* Make the parentRef different
* Make the call go to the Gateway OR to the service

I worry if we go down a path of duplication of every test we are going to struggle to keep them in sync, get parity, etc.

Reply:
I agree with you here. Thats why the linked issue mentions we need a framework change (to do exactly what you suggested) Over Kubecon EU though - @youngnick and @robscott pointed out there is preference to get in tests faster and hence to do a copy paste. To at least close the gap now (as gamma is GA) and follow up with the suggested change

Also so @kflynn can qualify linkerd with these tests.

Reply:
Yes, I'd definitely like to see us update the suite to reduce duplication, but as @LiorLieberman says, I'd rather see us get something in for 1.3 so that we can establish that we are interested in updating these tests and to get the pattern started  at least.